external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala

```diff
@@ -67,13 +67,18 @@ private[avro] class AvroFileFormat extends FileFormat
           spark.sessionState.conf.ignoreCorruptFiles)
     }
 
-    SchemaConverters.toSqlType(avroSchema).dataType match {
+    val schemaType = SchemaConverters.toSqlType(avroSchema)
+
+    schemaType.dataType match {
       case t: StructType => Some(t)
-      case _ => throw new RuntimeException(
-        s"""Avro schema cannot be converted to a Spark SQL StructType:
-           |
-           |${avroSchema.toString(true)}
-           |""".stripMargin)
+      case _ => Some(StructType(Seq(StructField("value", schemaType.dataType, nullable = false))))
```

Comment:
Wait, does this PR target to support to read values not the Avro records? Since Spark assumes rows, I think it's weird to allow values alone. I think JSON doesn't support this one. How did you generate the Avro files out of curiosity?

Reply:
Yes. This PR intends to support reading avro types other than records. We had a valid use case where upstream was generating these types and one of the downstream job in the pipelines was consuming it in spark. I just checked, from Spark 2.3, json doesn't support this one, so you are right. But I don't see a reason why not to.

You can generate avro data using the `avro-tools` jar available with avro. 
To generate random data you just specify the schema and the no. of records you want and it will generate the data for you. Example:
`java -jar avro-tools-1.8.2.jar random --count 20 --schema '{"type": "map", "values": "long"}' randomLongMap.avro`

If you haven't already used it, you will find it interesting. I have personally used it quite a few times for generating test datasets.

Reply:
@HyukjinKwon @cloud-fan any thoughts on the PR/feature?

Reply:
Even though we currently do not support it with json, IMHO it is still a useful feature.